homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -1,8 +1,12 @@
 """Support for Enphase Envoy solar energy monitor."""
+
+from datetime import timedelta
 import logging
 
+import async_timeout
 from envoy_reader.envoy_reader import EnvoyReader
-import requests
+import httpx
+from requests_async.exceptions import HTTPError, RequestException
```

Comment:
`requests_async` is deprecated. We should not use it in new code.  Please use the httpx exceptions intead.

https://github.com/encode/requests-async

Reply:
Replaced `requests_async` with `httpx`.

I also did this in the `envoy_reader` library so the version for the library has bumped to 0.17.2

Here is the [change log](https://github.com/jesserizzo/envoy_reader/compare/0.17.0...0.17.2) for the envoy_reader

Reply:
I bumped the envoy_reader  to `0.18.2` as after I moved from `requests_async` to `httpx`, I was seeing `httpx.RemoteProtocolError` exceptions from older Envoy devices. I added a retry in the library so it should hopefully eliminate or at least minimize the exceptions occurring on older devices.